### ISSUE TITLE
Add basic LDtk integration

### ISSUE THREAD

**Issue by keir** [deepnight/gameBase#25]
*tl;dr* - it's hard for a Haxe/Heaps newb to start with @deepnight 's tutorial on gameBase and end up with a LDtk level in their game.

This ticket could perhaps be split between LDtk and gameBase, but I'll put the bulk of the ticket here.

It's not obvious how to bring LDtk levels into gameBase. While there is an example in ld47 (Footloop), there is enough additional ancillary code in ld47 that it's hard to get going quickly. This is from someone with no Heaps or Haxe experience, so maybe this is obvious to others but just not me. The docs in LDtk explaining the Haxe API show how to load a LDtk level into Haxe objects, but don't explain how one might render it. For example, where are the assets loaded from? Is that handled in the background?

Suggested fixes:

- [ ] Extend LDtk docs to mention gameBase and how to integrate a LDtk level there.
- [ ] Extend gameBase to have a basic LDtk level (maybe by default)? Not just loading the level, but rendering it, maybe with 1 entity converted.
- [ ] Add a short note about things to consider when structuring the level / importer from the perspective of a gameBase user.
- [ ] Add a note somewhere about asset workflow for LDtk in the gameBase/Heaps context. Is there anything to know?

Thank you for releasing the wonderful LDtk tool!

**Comment by harshetmaruff** [comment]
Yes, I am also a beginner at haxe and heaps and its very hard to start for me, i see very few docs at heaps

**Comment by deepnight** [comment]
Hi, you can use the `advancedBase` branch to have a LDtk integration out-of-the-box.

I'll add more documentation asap :)

**Comment by keir** [comment]
I was able to make the LDtk integration work on my own before I saw the update about the advanced branch. However, I'm now stuck at the part that the gameBase advanced version doesn't cover: collisions. I took the code from the sample article and plugged it in, but it seems there is some scaling being applied to the entities that isn't applied to the level. It's not clear why this is happening. From what I can tell, my code is doing the same thing as the advanced gameBase and also ld47.

Here's my collision code in update():

```
		// X
		var steps = M.ceil( M.fabs(dxTotal*tmod) );
		var step = dxTotal*tmod / steps;
		while( steps>0 ) {
			xr+=step;

			if( level.hasCollision(cx+1,cy) && xr>=0.7 ) {
				trace("Hit level left!");
				xr = 0.7;
				dx *= Math.pow(0.5,tmod);
			}

			if( level.hasCollision(cx-1,cy) && xr<=0.3 ) {
				trace("Hit level right!");
				xr = 0.3;
				dx *= Math.pow(0.5,tmod);
			}

			while( xr>1 ) { xr--; cx++; }
			while( xr<0 ) { xr++; cx--; }
			steps--;
		}
```

I'll keep plugging away at it. Unfortunately there is enough unfamiliar stuff for me that this isn't the fastest!

**Comment by edwinsyarief** [comment]
Hi @deepnight ,

I'm using advanceBase as starter. Is there any example that I can change in game resolution when game is running?

Thank you

**Comment by deepnight** [comment]
Full LDtk integration in `GameBase->Advanced base` branch
